pkg/common/vclib/datacenter.go

```diff
@@ -77,6 +77,23 @@ func GetNumberOfDatacenters(ctx context.Context, connection *VSphereConnection)
 	return len(datacenters), nil
 }
 
+// GetVMByIP gets the VM object from the given IP address
+func (dc *Datacenter) GetVMByIP(ctx context.Context, ipAddy string) (*VirtualMachine, error) {
+	s := object.NewSearchIndex(dc.Client())
+	ipAddy = strings.ToLower(strings.TrimSpace(ipAddy))
+	svm, err := s.FindByIp(ctx, dc.Datacenter, ipAddy, true)
+	if err != nil {
+		klog.Errorf("Failed to find VM by IP. VM IP: %s, err: %+v", ipAddy, err)
+		return nil, err
+	}
+	if svm == nil {
+		klog.Errorf("Unable to find VM by IP. VM IP: %s", ipAddy)
+		return nil, ErrNoVMFound
+	}
+	virtualMachine := VirtualMachine{object.NewVirtualMachine(dc.Client(), svm.Reference()), dc}
```

Comment:
`FindByIp` returns interface type `Reference`, that'll be a `*object.VirtualMachine` so you can avoid `NewVirtualMachine`:
```go
virtualMachine := VirtualMachine{svm.(*object.VirtualMachine), dc}
```

Same goes for the other cases where CCM uses SearchIndex.FindBy{Uuid,DnsName}.. not a huge resource savings, but would make the code slightly more concise.  Up to you on making that change here or in a separate PR, or not at all.

Reply:
Updated!